components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/payment/ValidatePaymentCharge.java

```diff
@@ -128,6 +154,9 @@ public void validate(String json) throws CustomException {
             } catch (CustomException e){
                 log.error("Manipulating received JSON Object: " + e);
                 throw new CustomException(e.getErrcode(), e.getErrmsg(), e.getErrvar());
+            } catch (BadPaddingException e) {
+                log.error("Error occurred while unmasking. Possible reason would be incorrect masking configuration. " , e);
+                throw new CustomException("SVC0001", "A service error occurred.", new String[]{"Invalid user mask configuration"});
             } catch (Exception e) {
```

Comment:
[Code Review] Remove this exception if possible. Capture more specific exceptions instead.